apps/freeablo/faworld/gamelevel.cpp

```diff
@@ -125,15 +131,17 @@ namespace FAWorld
         actorMapInsert(actor);
     }
 
-    void GameLevel::fillRenderState(FARender::RenderState* state)
+    void GameLevel::fillRenderState(FARender::RenderState* state, const Player& player)
     {
         state->mObjects.clear();
 
         for(size_t i = 0; i < mActors.size(); i++)
         {
-            size_t frame = mActors[i]->mFrame + mActors[i]->mPos.getDirection() * mActors[i]->getCurrentAnim()->getAnimLength();
+            if (player.canSee(mActors[i]->mPos.current())) {
```

Comment:
Add a version that takes the actor as argument so we don't needlessly expose internals to another class (GameLevel)